java/flight/src/test/java/org/apache/arrow/flight/auth/TestAuth.java

```diff
@@ -94,22 +95,38 @@ public boolean isValid(byte[] token) {
       }
     };
 
-    server = new FlightServer(
-        allocator,
-        l.getPort(),
-        new NoOpFlightProducer() {
-          @Override
-          public void listFlights(Criteria criteria, StreamListener<FlightInfo> listener) {
-            listener.onCompleted();
-          }
-        },
-        new BasicServerAuthHandler(validator));
-
-    server.start();
-    client = new FlightClient(allocator, l);
-  }
-
+    // choose a port from the random/private range (https://en.wikipedia.org/wiki/Registered_port)
+    // we see flakiness when we use the same port each time.
+    Location location = null;
+    IOException lastThrown = null;
+    Random random = new Random();
+    for (int x = 0; x < 3; x++) {
+      final int port = 49152 + random.nextInt(5000);
+      location = new Location("localhost", port);
+      lastThrown = null;
+      try {
+        server = new FlightServer(
```

Comment:
do we need to start the server for each test? can we do it once for the class and create new clients for each test?

Reply:
In general I would prefer not to do that as long as server start is quick (which I think it should be).

Reply:
hmm ok..would that not be the actual real world scenario though? (should we have another test for that else?)

Reply:
Yes, I think another test would be better for that.  I haven't been involved in flight development but I'll leave up to the community members that have to open a JIRA if one already isn't open for more integration testing like scenarios.

Reply:
Ok i am not sure i agree :) but either ways the test is in much better shape than it was..we can refactor more later. 

Thanks again !